org.mixedrealitytoolkit.spatialmanipulation/ObjectManipulator/ManipulationSpecializations/BaseManipulationSpecialization.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace MixedReality.Toolkit.SpatialManipulation
+{
+    /// <summary>
+    /// Abstract base class from which all manipulation specialization behaviors derive.
+    /// </summary>
+    /// <seealso cref="IManipulationSpecialization"/>
+    public abstract class BaseManipulationSpecialization : MonoBehaviour, IManipulationSpecialization
```

Comment:
Does this need to be a `MonoBehaviour`? I don't see any Unity method callbacks in the base or the implementation.

Reply:
As discussed in meeting, I only decided to try use MonoBehaviour due to it easily providing user tweakable fields - like the Constraint system.  But may go back to just having this be a "type selected" system (non-MonoBehaviour) like ManipulationLogic.  Or see if there would be a less invasive way to provide for this type of thing all together (from discussion).